### PR TITLE
[face-detection] add keypoint detection

### PR DESCRIPTION
**Pull request by still-wait**
1.  add `SSDWithLmkLoss`.
2. add `blazeface_keypoint` config.
3.  add keypoint data augmentation.
4. add keypoint visual.

### CODE REVIEW

File: configs/face_detection/blazeface_keypoint.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    is_normalized: true
+  - !Permute {}
+  - !NormalizeImage
+    is_scale: false
+    mean: [104, 117, 123]
+    std: [127.502231, 127.502231, 127.502231]
+  batch_size: 16
+  use_process: true
+  worker_num: 8
+  memsize: 10G
```
Comment (jerrywgz) [inline configs/face_detection/blazeface_keypoint.yml:93]: 能否使用默认值？

File: ppdet/data/source/widerface.py [hunk 2]
```diff
@@ -62,30 +64,23 @@ def load_roidb_and_cname2cid(self):
...
-        file_lists = _load_file_list(txt_file)
+        file_lists = self._load_file_list(txt_file)
         cname2cid = widerface_label(self.with_background)
 
         for item in file_lists:
             im_fname = item[0]
             im_id = np.array([ct])
-            gt_bbox = np.zeros((len(item) - 2, 4), dtype=np.float32)
-            gt_class = np.ones((len(item) - 2, 1), dtype=np.int32)
+            gt_bbox = np.zeros((len(item) - 1, 4), dtype=np.float32)
```
Comment (jerrywgz) [inline ppdet/data/source/widerface.py:73]: why change the size of gt_bbox?

File: ppdet/data/source/widerface.py [hunk 3]
```diff
@@ -105,34 +103,60 @@ def load_roidb_and_cname2cid(self):
...
+                if w < 0 or h < 0:
+                    continue
+                xmin = max(0, xmin)
+                ymin = max(0, ymin)
+                xmax = xmin + w
+                ymax = ymin + h
+                gt_bbox = [xmin, ymin, xmax, ymax]
+                result_boxs.append(gt_bbox)
+                if self.with_lmk:
+                    assert len(split_str) > 18
```
Comment (jerrywgz) [inline ppdet/data/source/widerface.py:139]: need error message

File: ppdet/data/transform/operators.py [hunk 4]
```diff
@@ -912,6 +943,10 @@ def __call__(self, sample, context):
                 sample['gt_bbox'] = crop_bbox
                 sample['gt_class'] = crop_class
                 sample['gt_score'] = crop_score
+                if 'gt_keypoint' in sample.keys():
+                    sample['gt_keypoint'] = gt_keypoints[0]
+                    sample['keypoint_ignore'] = gt_keypoints[1]
+                #print(sample)
```
Comment (jerrywgz) [inline ppdet/data/transform/operators.py:949]: remove comment code

File: ppdet/data/transform/operators.py [hunk 5]
```diff
@@ -952,6 +995,10 @@ def __call__(self, sample, context):
...
```
Comment (jerrywgz) [inline ppdet/data/transform/operators.py:1001]: same as above

File: ppdet/data/transform/operators.py [hunk 6]
```diff
@@ -1672,3 +1730,69 @@ def __call__(self, sample, context=None):
         bbox[:, :2] = bbox[:, :2] + bbox[:, 2:4] / 2.
         sample['gt_bbox'] = bbox
         return sample
+
+
+@register_op
```
Comment (jerrywgz) [inline ppdet/data/transform/operators.py:1735]: is it necessary to merge into master?

Comment (qingqing01) [reply ppdet/data/transform/operators.py:1735]: This visualized code is helpful, can merge

File: ppdet/data/source/widerface.py (hunk 2 above)
Comment (still-wait) [reply ppdet/data/source/widerface.py:73]: Cleaned the code in `_load_file_list`, so dimension of gt_bbox decreased.

**Pull request closed without merging**